### ISSUE TITLE
Ch2. non-exhaustive struct

### ISSUE THREAD

**Issue by tylerhjones** [skerkour/black-hat-rust#53]
I have the exact function in https://github.com/skerkour/black-hat-rust/blob/main/ch_02/tricoder/src/subdomains.rs#L45
After cloning the repo, it will build with cargo build. But when I followed along in your book, I get 
```
error[E0639]: cannot create non-exhaustive struct using struct expression
  --> src/subdomains.rs:56:9
   |
56 | /         ResolverOpts {
57 | |             timeout: Duration::from_secs(4),
58 | |             ..Default::default()
59 | |         },
   | |_________^
```

Why does it work in your example but not for me? Have I missed some information from the book, require a specific crate version, or was there something omitted from the book I need to know?

**Comment by tylerhjones** [comment]
Im guessing this is some change in the crate and Im at a newer version.
I was able to resolve with
```
    let mut opts = ResolverOpts::default();
    opts.timeout = Duration::from_secs(5);
    
    let dns_resolver = Resolver::new(
        ResolverConfig::default(), 
        opts,
    )
```
instead of using the ..Defaults:: thing. Maybe using the above could be more future proof for the crate versions?

**Comment by skerkour** [comment]
Fixed in c3b5a98c9fc520b5177de4eb63836af285fbdcf6
